code/modules/mob/living/carbon/human/life.dm

```diff
@@ -485,6 +485,8 @@
 	if(on_fire)
 		var/thermal_protection = get_thermal_protection()
 
+		if(wear_suit && wear_suit.flags & AUTOEXTINGUISH)
+			wear_suit.auto_extinguish(src)

```

Comment:
That'd mean that having nobreathe would cure them of it too since this proc is only called if the species is breathing